### PR TITLE
fixes #1964

### LINKED ISSUE #1964: Download Estimator - bug waiting to happen?
### To Reproduce
Look at the generated HTML source of a step 3 download estimator

### What happens
Looking at the generated source we see similar too:
```
<div id="downloadEst-undefined-1441684507291">.......
```
The function called to build the div id:
```
getIdElementName: function(uuid) {
        return String.format("downloadEst-{0}-{1}", uuid, this.initTimestampString);
    },
```

### What should happen
uuid should not be undefined or the getIdElementName function should be changed

### PR DESCRIPTION
**Pull request by pmbohm**

### CODE REVIEW

File: web-app/js/portal/cart/BaseInjector.js [hunk 1]
```diff
@@ -66,6 +70,7 @@ Portal.cart.BaseInjector = Ext.extend(Object, {
 
     _addDownloadEstimate: function(dataCollection) {
 
+        dataCollection.uuid = this._getCollectionUuid(dataCollection);
```
Comment (danfruehauf) [inline web-app/js/portal/cart/BaseInjector.js]: I don't think it's a good idea to mutate the `dataCollection` object here.

`dataCollection.uuid` should be set when the `dataCollection` object is first instantiated.

File: web-app/js/portal/cart/BaseInjector.js [hunk 2]
```diff
@@ -25,6 +25,10 @@ Portal.cart.BaseInjector = Ext.extend(Object, {
         };
     },
 
+    _getCollectionUuid: function(collection) {
```
Comment (danfruehauf) [inline web-app/js/portal/cart/BaseInjector.js]: And this too, it's not the right place for that.

**Comment by pmbohm** [comment]
@danfruehauf Thanks i found the correct built in function to use

File: web-app/js/portal/cart/BaseInjector.js [hunk 3]
```diff
@@ -66,6 +66,7 @@ Portal.cart.BaseInjector = Ext.extend(Object, {
 
     _addDownloadEstimate: function(dataCollection) {
 
+
```
Comment (danfruehauf) [inline web-app/js/portal/cart/BaseInjector.js]: That newline can go.

Comment (danfruehauf) [inline web-app/js/portal/cart/BaseInjector.js]: I'll do it.

**Comment by danfruehauf** [comment]
Saw that, I'll merge after tests pass.

**Pull request merged**